src/main/kotlin/com/ampnet/auditornode/script/api/classes/HttpClient.kt

```diff
@@ -0,0 +1,121 @@
+package com.ampnet.auditornode.script.api.classes
+
+import com.ampnet.auditornode.model.error.EvaluationError.InvalidInputValueError
+import com.ampnet.auditornode.script.api.model.HttpCookie
+import com.ampnet.auditornode.script.api.model.HttpResponse
+import com.ampnet.auditornode.script.api.model.ListApi
+import com.ampnet.auditornode.script.api.model.MapApi
+import io.micronaut.http.HttpMethod
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.MediaType
+import io.micronaut.http.MutableHttpRequest
+import io.micronaut.http.client.BlockingHttpClient
+import io.micronaut.http.cookie.SameSite
+import org.graalvm.polyglot.HostAccess.Export
+import org.graalvm.polyglot.Value
+import javax.inject.Singleton
+
+@Singleton
+class HttpClient(private val blockingHttpClient: BlockingHttpClient) { // TODO add support for promises
+
+    @Export
+    @JvmOverloads
+    fun get(url: String, headers: Value? = null): HttpResponse {
+        return httpRequest(
+            headers = headers,
+            request = HttpRequest.GET(url),
+            methodCall = "HttpClient.get()",
+            argumentIndex = 1
+        )
+    }
+
+    @Export
+    @JvmOverloads
+    fun post(url: String, body: String = "", headers: Value? = null): HttpResponse {
+        return httpRequest(
+            headers = headers,
+            request = HttpRequest.POST(url, body).apply {
+                if (body.isNotEmpty()) {
+                    contentType(MediaType.APPLICATION_JSON)
+                }
+            },
+            methodCall = "HttpClient.post()",
+            argumentIndex = 2
+        )
+    }
+
+    @Export
+    @JvmOverloads
+    fun request(url: String, method: String, body: String = "", headers: Value? = null): HttpResponse {
+        val upperCaseMethod = method.toUpperCase()
+        val httpMethod = HttpMethod.parse(upperCaseMethod)
+        return httpRequest(
+            headers = headers,
+            request = HttpRequest.create<String>(httpMethod, url, upperCaseMethod).apply {
+                if (body.isNotEmpty()) {
+                    body(body)
+                    contentType(MediaType.APPLICATION_JSON)
+                }
+            },
+            methodCall = "HttpClient.request()",
+            argumentIndex = 3
+        )
+    }
+
+    private fun httpRequest(
+        headers: Value?,
+        request: MutableHttpRequest<String>,
+        methodCall: String,
+        argumentIndex: Int
+    ): HttpResponse {
+        if (headers != null) {
+            createHttpHeaders(headers, request, methodCall, argumentIndex)
+        }
```

Comment:
```suggestion
        headers?.let { createHttpHeaders(headers, request, methodCall, argumentIndex) }
```

Looks to me, there is no need for `createHttpHeaders` function, maybe put all inside this function.